crates/sui-analytics-indexer/src/lib.rs

```diff
@@ -459,13 +478,17 @@ pub async fn make_move_call_processor(
     metrics: AnalyticsMetrics,
 ) -> Result<Processor> {
     let handler: Box<dyn AnalyticsHandler<MoveCallEntry>> = Box::new(MoveCallHandler::new());
-    let (epoch, starting_checkpoint_seq_num) =
-        read_store_for_epoch_and_checkpoint(config.remote_store_config.clone(), FileType::MoveCall)
-            .await?;
+    let starting_checkpoint_seq_num = if let Some(starting_checkpoint_seq_num) =
+        config.starting_checkpoint_seq_num
+    {
+        starting_checkpoint_seq_num
+    } else {
+        read_store_for_checkpoint(config.remote_store_config.clone(), FileType::Checkpoint).await?
+    };
     let writer: Box<dyn AnalyticsWriter<MoveCallEntry>> = match config.file_format {
         FileFormat::CSV => Box::new(CSVWriter::new(
             &config.checkpoint_dir,
-            epoch,
+            0,
```

Comment:
nit: FWIU we rely on the following reset call to set the epoch properly. Is there any reason to keep this parameter in the constructor? (or should we change internal `epoch` field to `Option<u64>`)

Reply:
yeah good call, we can just make epoch initialize with 0 without making it a constructor argument